interface.go

```diff
@@ -68,8 +68,15 @@ type Interface interface {
 	Shutdown(ctx context.Context) error
 }
 
+// MultihashIterator iterates over a list of multihashes.
+type MultihashIterator interface {
+	// Next returns the next multihash in the list of mulitihashes.
+	// This function return io.EOF when there are no more multihashes to return.
```

Comment:
I'd say "a zero multihash and io.EOF" to clarify that EOF is never returned along with the last value.

See io.Reader, which *can* give you the last chunk of bytes along with EOF to save a call.